### ISSUE TITLE
Add a Timeline view

### ISSUE THREAD

**Issue by marcusolsson** [marcusolsson/obsidian-projects#6]
Add a Timeline view to lay out long-spanning calendar items in parallel. Similar to a Gantt chart.

**Comment by robynrbieber** [comment]
This would be amazing1

**Comment by marcusolsson** [comment]
Since there seems to be quite some interest in this kind of view, I'm bumping the priority.

**Comment by GamerGirlandCo** [comment]
heads up: i'm working on implementing this! <sub>only problem is that the code i wrote is a total trashfire...</sub>

**Comment by marcusolsson** [comment]
> heads up: i'm working on implementing this! only problem is that the code i wrote is a total trashfire...

Exciting! I imagine this require some time and effort, so please feel free to open a draft PR to get some feedback.

At a minimum I imagine you'd pick two fields: Start and End dates. I was planning to look at the [d3-scale](https://github.com/d3/d3-scale) which has a time scale for mapping a timestamp to a pixel, which would likely be useful here.

Let me know how it goes!

**Comment by BlackScorp** [comment]
obisidian has a built in mermaid support, which has a gantt chart https://mermaid.js.org/syntax/gantt.html  so maybe you could create markdown files with the gantt chart?

**Comment by marcusolsson** [comment]
I'll give an update here, since this is the most requested feature in the backlog.

While I'd love to have a Timeline view for my own projects, I won't personally prioritize working on any new views for a while. 

A few reasons:

- In an attempt to use my time more wisely, I'm trying to focus on cross-cutting bugs and features that affect all views.
- Introducing a new view introduces a whole new category of feature requests and bug reports specific to that view.
- Views are isolated parts of the plugin. Each view acts almost like a plugin within the Projects plugin. I believe this makes it easier for other contributors, as you can build new views and improve existing ones without having to understand the rest of the Projects codebase.

This is also a reminder that anyone can develop a view without having to fork the Projects plugin. You can use the [Custom View API](https://github.com/marcusolsson/obsidian-projects/tree/main/obsidian-projects-types) to build your own Projects view from your own plugin. Since the core views use the same API, it should be pretty easy to merge it as a core view down the line.

**Comment by viperbbb** [comment]
I was looking for this future, now I know still on the way !!!!!!!!

**Comment by amglez** [comment]
I would love to see it too
<3

**Comment by Ampa** [comment]
Please please please add a timeline view.
Would be so helpful to me.
Thanks.

**Comment by FREYA265** [comment]
Hello, first of all thank you for your work ^^ I love it!
Secondly, I think this is a great idea!
It's exactly what prompted me to create an account to exchange with you ^^
Yes, for example, I really like the Trello view! With the timeline.
So I'm really happy that you're working on this project ^^

![image](https://github.com/marcusolsson/obsidian-projects/assets/139784879/5d0b8ae6-979d-41b3-8e5d-bdeb45a18e3c)

**Comment by aiyou9** [comment]
an open source project of gantt:  https://github.com/worktile/ngx-gantt

playground: http://gantt.ngnice.com/components/basic

![image](https://github.com/marcusolsson/obsidian-projects/assets/15712822/825c8325-1e8e-4a1a-aee9-4ea93b8a33a2)

**Comment by roaldarbol** [comment]
Just found [Markwhen](https://markwhen.com/), which is an open source project that is also aiming at integrating with other IDEs/editors. They already have a VS Code extension, but importantly they also have Obsidian on the roadmap, which can be followed in https://github.com/mark-when/markwhen/issues/103! Might be worth keeping an eye on, as it could then hopefully be adapted to Projects down the line.

**Comment by beatblender** [comment]
I like your work very much! Thanks you very much! 
Can't wait to see your Gantt view 👍

**Comment by briandbezerra** [comment]
I can't imagine how hard it is to maintain, kills bugs, etc. However, this feature seems indispensable for, well, projects. I was surprised 'Projects' didn't have a 'spanning calendar' feature as it seems essential if one intends to use it for project management.

This plugin is superb and with this feature it would make it indispensable for me, and given the votes for this feature, a lot of other people. I wish this was prioritized (and that I actually knew what I was doing to help).

**Comment by Shaddy-ElDesouky** [comment]
upvote
I'll try mermaid plugin @BlackScorp  , and watch for Markwhen @roaldarbol  , useful thread, thanks

**Comment by HilkopterBob** [comment]
Any Updates @Acylation ?

**Comment by camilogarciabotero** [comment]
There is a quite recent alternative plugin for timelines: [markwhen](https://github.com/mark-when/obsidian-plugin)

![](https://github.com/mark-when/obsidian-plugin/blob/main/screenshot.png?raw=true)

**Comment by obstschale** [comment]
@emendir I am using Markwhen for few weeks now and it works really nice. The WebUI is slightly better, but the obsidian plugin is currently under development.

I was also thinking about programming an obsidian plugin, which simply grabs frontmatter values and convert them to markwhen syntax. This seems not that difficult, however all files need to have the same frontmatter fields. 

Unfortunately, I am currently not able to start (another) side project. :(

**Comment by Acylation** [comment]
Yeah I’m also on the markwhen thread. Feel free to find me there.

About the idea to grab front matter times to a single markwhen file, could templater help over this case?

Experimenting with some essential components to build timeline views for Projects. Still a long way to go.
